docs/standard/basetypes/alternation.md

```diff
@@ -0,0 +1,328 @@
+* Pattern matching with **|**
+
+* Conditional matching with **(?(**_expression_**)**_yes_**|**_no_**)**
+
+* Conditional matching based on a valid captured group
+
+## Pattern Matching with |
+
+You can use the vertical bar (|) character to match any one of a series of patterns, where the | character separates each pattern. 
+
+Like the positive character class, the | character can be used to match any one of a number of single characters. The following example uses both a positive character class and either/or pattern matching with the | character to locate occurrences of the words "gray" or "grey" in a string. In this case, the | character produces a regular expression that is more verbose. 
+
+```csharp
+using System;
+using System.Text.RegularExpressions;
+
+public class Example
+{
+   public static void Main()
+   {
+      // Regular expression using character class.
+      string pattern1 = @"\bgr[ae]y\b";
+      // Regular expression using either/or.
+      string pattern2 = @"\bgr(a|e)y\b";
+
+      string input = "The gray wolf blended in among the grey rocks.";
+      foreach (Match match in Regex.Matches(input, pattern1))
+         Console.WriteLine("'{0}' found at position {1}", 
+                           match.Value, match.Index);
+      Console.WriteLine();
+      foreach (Match match in Regex.Matches(input, pattern2))
+         Console.WriteLine("'{0}' found at position {1}", 
+                           match.Value, match.Index);
+   }
+}
+// The example displays the following output:
+//       'gray' found at position 4
+//       'grey' found at position 35
+//       
+//       'gray' found at position 4
+//       'grey' found at position 35           
+```
+
+```vb
+Imports System.Text.RegularExpressions
+
+Module Example
+   Public Sub Main()
+      ' Regular expression using character class.
+      Dim pattern1 As String = "\bgr[ae]y\b"
+      ' Regular expression using either/or.
+      Dim pattern2 As String = "\bgr(a|e)y\b"
+
+      Dim input As String = "The gray wolf blended in among the grey rocks."
+      For Each match As Match In Regex.Matches(input, pattern1)
+         Console.WriteLine("'{0}' found at position {1}", _
+                           match.Value, match.Index)
+      Next      
+      Console.WriteLine()
+      For Each match As Match In Regex.Matches(input, pattern2)
+         Console.WriteLine("'{0}' found at position {1}", _
+                           match.Value, match.Index)
+      Next      
+   End Sub
+End Module
+' The example displays the following output:
+'       'gray' found at position 4
+'       'grey' found at position 35
+'       
+'       'gray' found at position 4
+'       'grey' found at position 35 
+```
+
+The regular expression that uses the | character, `\bgr(a|e)y\b,` is interpreted as shown in the following table.
+
+Pattern | Description
+------- | -----------
+`\b` | Start at a word boundary.
+`gr` | Match the characters "gr".
+`(a&#124;e)` | Match either an "a" or an "e".
```

Comment:
you cannot encode pipes inside inline code, it'll not render correctly. you need to use HTML instead. F# content has a few examples of this.